### PR TITLE
feat: document the artifacts cache configuration settings

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none
Ticket: MEN-5316

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md [hunk 1]
```diff
@@ -44,8 +48,14 @@ provides a reference for the configuration variables.
 ```
 
 ### Features
-#### ArtifactProxy
 
+#### ArtifactsCache
+<dl>
+<dt>Enabled</dt> <dd>Enable the Artifact Cache (depends on `ArtifactsProxy`).</dd>
```
Comment (merlin-northern) [inline 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md:54]: I would mention "see below" or provide a link to the `ArtifactsProxy` section.
idle comment: this file is the only one we use <dl>, <dt>, <dd> tags, is there a reason to it?

File: 08.Server-integration/04.Mender-Gateway/docs.md [hunk 2]
```diff
@@ -23,11 +23,12 @@ authentication requests on behalf of devices.
 
 ![Mender gateway](mender-gateway-schema.png)
 
-## Artifact Proxy
+## Artifact Proxy and Cache
 The Mender Gateway is able to understand when a device has an available update
-and is able to serve artifacts on behalf of the server. Configuring Mender
+and is able to serve artifacts on behalf of the server. It can also cache the
```
Comment (merlin-northern) [inline 08.Server-integration/04.Mender-Gateway/docs.md:28]: nit: "is able" is repeated here, perhaps change here: "and is able to serve artifacts" -> "and _can_ serve artifacts"

File: 08.Server-integration/04.Mender-Gateway/docs.md [hunk 3]
```diff
@@ -23,11 +23,12 @@ authentication requests on behalf of devices.
...
+artifacts locally and serve them locally when needed. Configuring the Mender
 Gateway to serve as an artifact proxy will make the gateway serve any artifacts
-assigned by a deployment. See the [Artifact Proxy User Guide](../../01.Get-started/06.Mender-Gateway/)
+assigned by a deployment. See the [Mender Gateway User Guide](../../01.Get-started/06.Mender-Gateway/)
```
Comment (merlin-northern) [inline 08.Server-integration/04.Mender-Gateway/docs.md:31]: nit: usually we link to a certain file. shall we point to a certain place in the `../../01.Get-started/06.Mender-Gateway/`

**Review by merlin-northern — APPROVED** [review]
looks good, left some comments.

File: 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md (hunk 1 above)
Comment (tranchitella) [reply 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md:54]: I added the link. I don't know why this is the only file where we use `dl/dt/dd`.

File: 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md [hunk 4]
```diff
@@ -12,6 +12,10 @@ provides a reference for the configuration variables.
 ```json
 {
     "Features": {
+        "ArtifactsCache": {
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md:15]: I'm okay with ArtifactCache, but won't it be confusing to have ArtifactCache and ArtifactsProxy?

File: 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md [hunk 5]
```diff
@@ -44,8 +48,14 @@ provides a reference for the configuration variables.
 ```
 
 ### Features
-#### ArtifactProxy
 
+#### ArtifactsCache
+<dl>
+<dt>Enabled</dt> <dd>Enable the Artifact Cache (depends on [ArtifactsProxy](#artifactproxy)).</dd>
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md:54]: Current implementation is plural (ArtifactsProxy).

File: 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md [hunk 6]
```diff
@@ -89,7 +99,6 @@ This user will preauthorize devices with authorization to the gateway.
 </dl>
 
 ### UpstreamServer
-
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md:92]: To make it consistent with the other sections

File: 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tranchitella) [reply 08.Server-integration/04.Mender-Gateway/99.Configuration-file/docs.md:15]: @oleorhagen I updated the ticket using the plural version to make it consistent with "ArtifactsProxy".

File: 08.Server-integration/04.Mender-Gateway/docs.md [hunk 7]
```diff
@@ -23,16 +23,17 @@ authentication requests on behalf of devices.
...
-## Artifact Proxy
-The Mender Gateway is able to understand when a device has an available update
-and is able to serve artifacts on behalf of the server. Configuring Mender
-Gateway to serve as an artifact proxy will make the gateway serve any artifacts
-assigned by a deployment. See the [Artifact Proxy User Guide](../../01.Get-started/06.Mender-Gateway/)
+## Artifact Proxy and Cache
+The Mender Gateway can understand when a device has an available update
+and serves artifacts on the server's behalf. It can also cache the artifacts
+locally and serve them locally when needed. Configuring the Mender Gateway to
+serve as an artifact proxy will make the gateway serve any artifacts assigned
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 08.Server-integration/04.Mender-Gateway/docs.md]: @oleorhagen what about now?

**Review by oleorhagen — APPROVED** [review]
Nice :)

Pre-approving. Just remember we uppercase `Artifact`.

File: 08.Server-integration/04.Mender-Gateway/docs.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (oleorhagen) [reply 08.Server-integration/04.Mender-Gateway/docs.md]: Nice :smile:

File: 08.Server-integration/04.Mender-Gateway/docs.md [hunk 8]
```diff
@@ -23,16 +23,22 @@ authentication requests on behalf of devices.
...
-assigned by a deployment. See the [Artifact Proxy User Guide](../../01.Get-started/06.Mender-Gateway/)
+## Artifact Proxy and Cache
+The Mender Gateway can understand when a device has an available update
+and serves artifacts on the server's behalf. Without this feature, the device
+would try downloading the artifacts referenced by the download link contained
+in the deployment instructions directly from the internet. The Mender Gateway
+will instead replace the download link in the deployment instructions with a
+local URL served by itself. It can also cache the artifacts locally and serve
+them locally when needed. In this way, when multiple devices request the same
+artifact, it will be downloaded only once and then served from the cache for
```
Comment (oleorhagen) [inline 08.Server-integration/04.Mender-Gateway/docs.md:34]: Like here

Comment (tranchitella) [reply 08.Server-integration/04.Mender-Gateway/docs.md]: Done with a different commit.

**Pull request merged**